### DISCUSSION TITLE
Pass an RTL-SDR to LXC container

### DISCUSSION THREAD

**Discussion by optimist555** [merbanan/rtl_433#3088] [Q&A]
I have successfully run rtl_433 in a unprivileged LXC container on Proxmox and passed through the `/dev/bus/usb/***/***` path.
I use the function in the Proxmox GUI for LXC continers " Device passthrough". When I enter the full device path, rtl_433 finds the SDR and starts scanning right away.

There is a problem though, that the device path changes when changing USB port.
I have set up a `udev `rule to automatically create a symlink from device path to `/dev/rtl_sdr` using the following rule:
`SUBSYSTEM=="usb", ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="2838", MODE="0666", SYMLINK+="rtl_sdr"`

This successfully updates the symlink every time the server reboots or device unplugs.
But I can not find a way to specify a custom device path for rtl_433 so I can set it to use this symlink instead of autodetecting USB devices. If I start the container with the symlink passed through rtl_433 cannot find the SDR.

Is there a way to set a custom device path in the `rtl_433.conf`? I want to avoid running the container privileged.

**Comment by zuckschwerdt** [comment]
rtl_433 does not use usb devices or paths. It relies on librtlsdr which in turn uses libusb, e.g. https://github.com/osmocom/rtl-sdr/blob/master/src/librtlsdr.c#L1310
You need to have a passed-through device that is recognised by libusb.
